File: Day 1/2/count_vowels.c
#include <stdio.h>
#include <string.h>

int countVowels(const char *s) {
    int count = 0;
    for (int i = 0; s[i] != '\0'; i++) {
        char c = s[i];
        if (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u') {
            count++;
        }
    }
    return count;
}

int main() {
    char s[1001];

    printf("Enter a string: ");

    if (fgets(s, sizeof(s), stdin)) {
        s[strcspn(s, "\n")] = '\0';
    }

    printf("Number of vowels: %d\n", countVowels(s));
    
    return 0;
}

File: Day 1/2/countVowels.js
const countVowels = (s) => {
    const vowels = "aeiou";
    return s.split("").filter(char => vowels.includes(char)).length;
}

function countVowelsAlgorithmically(s) {
    let count = 0;
    const vowels = "aeiou";
    for (let i = 0; i < s.length; i++) {
        for (let j = 0; j < vowels.length; j++) {
            if (s[i] === vowels[j]) {
                count++;
                break;
            }
        }
    }
    return count;
}

console.log(countVowels("hello"));
console.log(countVowelsAlgorithmically("hello"));